Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentEnd/FindInPage.js

```diff
@@ -5,277 +5,320 @@
+  webkit.messageHandlers.findInPageHandler.postMessage({ currentResult: 0, totalResults: 0 });
 }
 
 function updateActiveHighlight() {
-  // Reset the color of the previous highlight.
-  if (activeHighlightSpan) {
-    activeHighlightSpan.style.backgroundColor = MATCH_HIGHLIGHT_INACTIVE;
+  if (!styleElement.parentNode) {
+    document.body.appendChild(styleElement);
   }
 
-  if (!highlightSpans.length) {
+  let lastActiveHighlight = document.querySelector("." + HIGHLIGHT_CLASS_NAME_ACTIVE);
+  if (lastActiveHighlight) {
+    lastActiveHighlight.className = HIGHLIGHT_CLASS_NAME;
+  }
+
+  if (!lastHighlights) {
     return;
   }
 
-  activeHighlightSpan = highlightSpans[activeIndex];
-  activeHighlightSpan.style.backgroundColor = MATCH_HIGHLIGHT_ACTIVE;
-
-  // Find the position of the element centered on the screen, then scroll to it.
-  var rect = activeHighlightSpan.getBoundingClientRect();
-  var top = SCROLL_OFFSET + rect.top + scrollY - window.innerHeight / 2;
-  var left = rect.left + scrollX - window.innerWidth / 2;
-  left = clamp(left, 0, document.body.scrollWidth);
-  top = clamp(top, 0, document.body.scrollHeight);
-  scrollToSelection(left, top, SCROLL_INTERVAL_DURATION);
-  debug("Scrolled to: " + left + ", " + top);
+  let activeHighlight = lastHighlights[activeHighlightIndex];
+  if (activeHighlight) {
+    activeHighlight.className = HIGHLIGHT_CLASS_NAME + " " + HIGHLIGHT_CLASS_NAME_ACTIVE;
+    scrollToElement(activeHighlight);
+
+    webkit.messageHandlers.findInPageHandler.postMessage({ currentResult: activeHighlightIndex + 1 });
+  }
 }
 
-function scrollToSelection(left, top, duration) {
-  var time = 0;
-  var startX = scrollX;
-  var startY = scrollY;
-  clearInterval(scrollInterval);
-  scrollInterval = setInterval(function() {
-    var xStep = easeOutCubic(time, startX, left - startX, duration);
-    var yStep = easeOutCubic(time, startY, top - startY, duration);
-    window.scrollTo(xStep, yStep);
-    time += SCROLL_INTERVAL_INCREMENT;
-    if (time >= duration) {
-      clearInterval(scrollInterval);
+function removeHighlight(highlight) {
+  let parent = highlight.parentNode;
+  if (parent) {
+    while (highlight.firstChild) {
+      parent.insertBefore(highlight.firstChild, highlight);
     }
-  }, SCROLL_INTERVAL_INCREMENT);
-}
 
-function easeOutCubic(currentTime, startValue, changeInValue, duration) {
-  return changeInValue * (Math.pow(currentTime / duration - 1, 3) + 1) + startValue;
+    highlight.remove();
+    parent.normalize();
+  }
 }
 
-function clamp(number, min, max) {
-  return Math.max(min, Math.min(number, max));
+function asyncTextNodeWalker(iterator, callback) {
+  let operation = new Operation();
+  let walker = document.createTreeWalker(document.body, NodeFilter.SHOW_TEXT, null, false);
+
+  let timeout = setTimeout(function() {
+    chunkedLoop(function() { return walker.nextNode(); }, function(node) {
+      if (operation.cancelled) {
+        return false;
+      }
+
+      iterator(node);
```

Comment:
there is `return false` above, should this return something also?

Reply:
We explicitly look for `=== false`, so, its not necessary.